search/lib/validator/index.js

```diff
@@ -0,0 +1,43 @@
+const fs = require('fs');
+const yaml = require('js-yaml');
+const path = require('path');
+const Ajv = require('ajv');
+const ajv = new Ajv({ allErrors: true });
+
+const loadOpenApiYaml = (swaggerYaml) => {
+  if (!swaggerYaml) throw new Error('Missing Yaml path');
+  let yamlSchemaFile;
+  if (path.isAbsolute(swaggerYaml)) {
+    yamlSchemaFile = swaggerYaml;
+  } else {
+    yamlSchemaFile = path.join(__dirname, swaggerYaml);
+  }
+
+  return yaml.safeLoad(fs.readFileSync(yamlSchemaFile));
+};
+
+const getSchema = (schemaName) => {
```

Comment:
I am not sure what the goal of this function is, could you please explain?

Reply:
I think the idea was for this function to get the list of components from the schema. `loadOpenApiYaml` took the schema file and converted into a js object, then this one went through and returns the 'components' nested object to be used in the `createSchemaValidator` function.

Reply:
could we make this function get the component by name from that collection? for instance, pass in the components object into the function with a name and get the schema from it.